app/models/airport.rb

```diff
@@ -0,0 +1,9 @@
+class Airport < ApplicationRecord
+  belongs_to :city
+  has_many :departure_flights, dependent: :destroy
+  has_many :departures, through: :departure_flights, source: :flight
+  has_many :arrival_flights, dependent: :destroy
+  has_many :arrivals, through: :arrival_flights, source: :flight
```

Comment:
I don't think you need a N:N relation between airports and flights and hence, you don't need the departure_flights and arrival_flights join tables.

A flight belongs to a departure_airport, and belongs to an arrival_airport, and an airport has many departure_flights and has many arrival_flights. You can model this as 2 1:N relationships between airports and flights

Reply:
Right, nice one! so this should be the right way:

<img width="340" alt="image" src="https://github.com/ImMPrada/TOP-flights-booking/assets/26731448/8bcda380-ef75-4aac-9e4b-6e5b453f6ebb">